app/controllers/submissions_controller.rb

```diff
@@ -491,6 +491,46 @@ def resubmit_to_turnitin
     end
   end
 
+   def vericite_report
+    return render(:nothing => true, :status => 400) unless params_are_integers?(:assignment_id, :submission_id)
+
+    @assignment = @context.assignments.active.find(params[:assignment_id])
+    @submission = @assignment.submissions.where(user_id: params[:submission_id]).first
+    @asset_string = params[:asset_string]
+    if authorized_action(@submission, @current_user, :read)
+      # vericite_data is a function, so store the results in an array to prevent multiple calls
+      vericite_data = @submission.vericite_data
+      if (report_url = vericite_data[@asset_string] && vericite_data[@asset_string][:report_url])
+        url = polymorphic_url([:retrieve, @context, :external_tools], url:report_url, display:'borderless')
+      else
+        url = @submission.vericite_report_url(@asset_string, @current_user) rescue nil
+      end
+      if url
+        redirect_to url
+      else
+        flash[:notice] = t('errors.no_report', "Couldn't find a report for that submission item")
+        redirect_to named_context_url(@context, :context_assignment_submission_url, @assignment.id, @submission.user_id)
+      end
+    end
+  end
+
+  def resubmit_to_vericite
```

Comment:
Similar to above, this code is largely the same so should be combined with the turn it in.